scrape/scrape_test.go

```diff
@@ -2384,9 +2387,68 @@ func TestTargetScraperScrapeOK(t *testing.T) {
 		require.Equal(t, "metric_a 1\nmetric_b 2\n", buf.String())
 	}
 
-	runTest(acceptHeader(config.DefaultScrapeProtocols))
+	runTest(acceptHeader(config.DefaultScrapeProtocols, false))
 	protobufParsing = true
-	runTest(acceptHeader(config.DefaultProtoFirstScrapeProtocols))
+	runTest(acceptHeader(config.DefaultProtoFirstScrapeProtocols, false))
+}
+
+func TestUTF8TargetScraperScrapeOK(t *testing.T) {
```

Comment:
I suggest merging this test with `TestTargetScraperScrapeOK` if possible. You can use booleans similar to `protobufParsing` in `TestTargetScraperScrapeOK` to determine if you should have the utf8 header or not when checking.